src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -256,9 +260,13 @@ bool CoreWorkerMemoryStore::Put(const RayObject &object, const ObjectID &object_
   }
 
   // It's important for performance to run the callbacks outside the lock.
-  for (const auto &cb : async_callbacks) {
-    cb(object_entry);
-  }
+  io_context_.post(
```

Comment:
this doesn't need to be posted since we are already out of `GetAsync`?

Reply:
Yes, actually this is the fix point of the original issue. Problem:

- caller holds `mu_`
- caller calls `Put()`
- `Put()` triggers callback
- callback try to hold `mu_` and deadlocks.

so we must post.